executor/src/witgen/jit/witgen_inference.rs

```diff
@@ -77,10 +77,14 @@ impl<'a, T: FieldElement, FixedEval: FixedEvaluator<T>> WitgenInference<'a, T, F
         }
     }
 
-    pub fn code(self) -> Vec<Effect<T, Variable>> {
+    pub fn finish(self) -> Vec<Effect<T, Variable>> {
         self.code
     }
 
+    pub fn code(&self) -> &[Effect<T, Variable>] {
```

Comment:
Where do we need this where we cannot use `finish()`?

Reply:
The reason for not passing owership of witgen to `solve_block` is thin, but I won't argue more here...